test/functional/rpc_orphans.py

```diff
@@ -105,6 +116,9 @@ def test_orphan_details(self):
         assert_equal(len(node.getorphantxs()), 1)
         orphan_1 = orphanage[0]
         self.orphan_details_match(orphan_1, tx_child_1, verbosity=1)
+        self.log.info("Checking orphan entry/expiration times")
+        assert_approx(orphan_1["entry"], entry_time, 5)
```

Comment:
Yeah, if we can, that would be simpler/preference.

Would need the overhead (message sending, processing, orphanage addition, etc.) to always take less than a second, right?  Otherwise there might be intermittent test failure.

If it's always under a second, then maybe something like `setmocktime(int(time.time()))` to force time to start at the beginning of a second might work?

Locally, I very roughly checked runtime and it was consistently under 105ms (not sure how accurate logger is though).  Any good way to benchmark on a less optimistic environment (e.g. congested CI)?

```diff
diff --git a/test/functional/rpc_orphans.py b/test/functional/rpc_orphans.py
index 207647f548..294a73066b 100755
--- a/test/functional/rpc_orphans.py
+++ b/test/functional/rpc_orphans.py
@@ -96,11 +96,14 @@ class OrphanRPCsTest(BitcoinTestFramework):
         tx_child_2 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_2["new_utxo"])
         peer_1 = node.add_p2p_connection(P2PInterface())
         peer_2 = node.add_p2p_connection(P2PInterface())
+        self.log.info("Setting time to the start of a second")
         entry_time = int(time.time())
+        node.setmocktime(entry_time)
         peer_1.send_and_ping(msg_tx(tx_child_1["tx"]))
         peer_2.send_and_ping(msg_tx(tx_child_2["tx"]))
 
         orphanage = node.getorphantxs(verbosity=2)
+        self.log.info(f"Just after first getorphantxs, entry_time={entry_time}")
         assert tx_in_orphanage(node, tx_child_1["tx"])
         assert tx_in_orphanage(node, tx_child_2["tx"])
```

```
2024-10-07T11:18:24.529000Z TestFramework (INFO): Setting time to the start of a second
2024-10-07T11:18:24.632000Z TestFramework (INFO): Just after first getorphantxs, entry_time=1728299904
```

Reply:
Thanks.

The following passed.  Will push.

```diff
diff --git a/test/functional/rpc_orphans.py b/test/functional/rpc_orphans.py
index 207647f548..825657a1ed 100755
--- a/test/functional/rpc_orphans.py
+++ b/test/functional/rpc_orphans.py
@@ -96,11 +96,15 @@ class OrphanRPCsTest(BitcoinTestFramework):
         tx_child_2 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_2["new_utxo"])
         peer_1 = node.add_p2p_connection(P2PInterface())
         peer_2 = node.add_p2p_connection(P2PInterface())
+        self.log.info("Setting time to the start of a second")
         entry_time = int(time.time())
+        node.setmocktime(entry_time)
+        time.sleep(3)
         peer_1.send_and_ping(msg_tx(tx_child_1["tx"]))
         peer_2.send_and_ping(msg_tx(tx_child_2["tx"]))
 
         orphanage = node.getorphantxs(verbosity=2)
+        self.log.info(f"Just after first getorphantxs, entry_time={entry_time}")
         assert tx_in_orphanage(node, tx_child_1["tx"])
         assert tx_in_orphanage(node, tx_child_2["tx"])
 
@@ -117,8 +121,8 @@ class OrphanRPCsTest(BitcoinTestFramework):
         orphan_1 = orphanage[0]
         self.orphan_details_match(orphan_1, tx_child_1, verbosity=1)
         self.log.info("Checking orphan entry/expiration times")
-        assert_approx(orphan_1["entry"], entry_time, 5)
-        assert_approx(orphan_1["expiration"], entry_time + ORPHAN_MAX_RETENTION_TIME, 5)
+        assert_equal(orphan_1["entry"], entry_time)
+        assert_equal(orphan_1["expiration"], entry_time + ORPHAN_MAX_RETENTION_TIME)
 
         self.log.info("Checking orphan details (verbosity 2)")
         orphanage = node.getorphantxs(verbosity=2)
```